<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>Fractal-Pattern-Generator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
       <normaloff>:/FPGresources/icons240/icons8-python.png</normaloff>:/FPGresources/icons240/icons8-python.png
   </iconset>
  </property>
  <widget class="QWidget" name="central_widget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGraphicsView" name="graphics_View"/>
    </item>
   </layout>
  </widget>
  <widget class="QDockWidget" name="dock_Widget">
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Меню</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dock_Widget_Contents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QFormLayout" name="form_layout">
       <item row="0" column="0">
        <widget class="QLabel" name="label_line_width">
         <property name="text">
          <string>Толщина</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="line_width">
         <property name="maximum">
          <double>1000.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Глубина</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="number_of_iter"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Длинна</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="line_length">
         <property name="maximum">
          <double>10000.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Угол поворота</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QDoubleSpinBox" name="rotation_angle">
         <property name="maximum">
          <double>10000.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Отклонение длинны</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QDoubleSpinBox" name="length_deviation">
         <property name="maximum">
          <double>10000.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Отклонение угла</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QDoubleSpinBox" name="rotation_angle_deviation">
         <property name="maximum">
          <double>10000.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Аксиома</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLineEdit" name="axiom">
           <property name="clearButtonEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="text_editing">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Правила</string>
       </property>
       <layout class="QVBoxLayout" name="_3">
        <item>
         <widget class="MText_list_Widget" name="rules_list">
          <property name="contextMenuPolicy">
           <enum>Qt::ActionsContextMenu</enum>
          </property>
          <property name="dragDropMode">
           <enum>QAbstractItemView::InternalMove</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Цвета</string>
       </property>
       <layout class="QVBoxLayout" name="_2">
        <item>
         <widget class="MColor_list_Widget" name="colors_list">
          <property name="contextMenuPolicy">
           <enum>Qt::ActionsContextMenu</enum>
          </property>
          <property name="dragDropMode">
           <enum>QAbstractItemView::InternalMove</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="start_button">
       <property name="text">
        <string>Старт</string>
       </property>
       <property name="icon">
        <iconset resource="resources.qrc">
            <normaloff>:/FPGresources/icons48/icons8-start.png</normaloff>:/FPGresources/icons48/icons8-start.png
        </iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="stop_button">
       <property name="text">
        <string>Стоп</string>
       </property>
       <property name="icon">
        <iconset resource="resources.qrc">
            <normaloff>:/FPGresources/icons48/icons8-stop.png</normaloff>:/FPGresources/icons48/icons8-stop.png
        </iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="clear_button">
       <property name="text">
        <string>Очистить</string>
       </property>
       <property name="icon">
        <iconset resource="resources.qrc">
            <normaloff>:/FPGresources/icons48/icons8-clean.png</normaloff>:/FPGresources/icons48/icons8-clean.png
        </iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
        <width>580</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="file_menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <widget class="QMenu" name="menu_3">
     <property name="title">
      <string>Недавно открытые</string>
     </property>
     <property name="icon">
      <iconset resource="resources.qrc">
          <normaloff>:/FPGresources/icons96/icons8-history-folder.png</normaloff>
          :/FPGresources/icons96/icons8-history-folder.png
      </iconset>
     </property>
    </widget>
    <addaction name="action_new_file"/>
    <addaction name="action_open_file"/>
    <addaction name="menu_3"/>
    <addaction name="separator"/>
    <addaction name="action_save_file"/>
    <addaction name="action_save_file_as"/>
    <addaction name="action_save_image"/>
    <addaction name="separator"/>
    <addaction name="action_escape"/>
   </widget>
   <widget class="QMenu" name="about_menu">
    <property name="title">
     <string>Справка</string>
    </property>
    <addaction name="action_about_Qt"/>
    <addaction name="action_about_program"/>
   </widget>
   <addaction name="file_menu"/>
   <addaction name="about_menu"/>
  </widget>
  <widget class="QToolBar" name="file_tools">
   <property name="windowTitle">
    <string>file_tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_new_file_2"/>
   <addaction name="action_open_file_2"/>
   <addaction name="action_save_file_2"/>
  </widget>
  <widget class="QToolBar" name="canvas_tools">
   <property name="windowTitle">
    <string>canvas_tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_zoom_in"/>
   <addaction name="action_zoom_out"/>
   <addaction name="action_reset_zoom_to_actual_size"/>
  </widget>
  <action name="action_zoom_in">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-zoom-in.png</normaloff>:/FPGresources/icons48/icons8-zoom-in.png
    </iconset>
   </property>
   <property name="text">
    <string>Приблизить</string>
   </property>
  </action>
  <action name="action_zoom_out">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-zoom-out.png</normaloff>:/FPGresources/icons48/icons8-zoom-out.png
    </iconset>
   </property>
   <property name="text">
    <string>Отдалить</string>
   </property>
  </action>
  <action name="action_reset_zoom_to_actual_size">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-zoom-to-actual-size.png</normaloff>
        :/FPGresources/icons48/icons8-zoom-to-actual-size.png
    </iconset>
   </property>
   <property name="text">
    <string>Масштабирование до фактического размера</string>
   </property>
  </action>
  <action name="action_open_file">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-opened-folder.png</normaloff>
        :/FPGresources/icons48/icons8-opened-folder.png
    </iconset>
   </property>
   <property name="text">
    <string>Открыть</string>
   </property>
  </action>
  <action name="action_save_file">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-save-as.png</normaloff>:/FPGresources/icons48/icons8-save-as.png
    </iconset>
   </property>
   <property name="text">
    <string>Сохранить</string>
   </property>
  </action>
  <action name="action_save_file_as">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-save-as.png</normaloff>:/FPGresources/icons48/icons8-save-as.png
    </iconset>
   </property>
   <property name="text">
    <string>Сохранить как...</string>
   </property>
  </action>
  <action name="action_save_image">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-image.png</normaloff>:/FPGresources/icons48/icons8-image.png
    </iconset>
   </property>
   <property name="text">
    <string>Сохрнить изображение</string>
   </property>
  </action>
  <action name="action_escape">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-logout.png</normaloff>:/FPGresources/icons48/icons8-logout.png
    </iconset>
   </property>
   <property name="text">
    <string>Выход</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="action_about_Qt">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/iconsQt/Qt_logo_2022.png</normaloff>:/FPGresources/iconsQt/Qt_logo_2022.png
    </iconset>
   </property>
   <property name="text">
    <string>О Qt</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutQtRole</enum>
   </property>
  </action>
  <action name="action_new_file">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-add-file.png</normaloff>:/FPGresources/icons48/icons8-add-file.png
    </iconset>
   </property>
   <property name="text">
    <string>Новый</string>
   </property>
  </action>
  <action name="action_new_file_2">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-add-file.png</normaloff>:/FPGresources/icons48/icons8-add-file.png
    </iconset>
   </property>
   <property name="text">
    <string>Новый файл</string>
   </property>
   <property name="toolTip">
    <string>Новый файл</string>
   </property>
  </action>
  <action name="action_open_file_2">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons240/icons8-opened-folder-240.png</normaloff>
        :/FPGresources/icons240/icons8-opened-folder-240.png
    </iconset>
   </property>
   <property name="text">
    <string>Открыть файл</string>
   </property>
   <property name="toolTip">
    <string>Открыть файл</string>
   </property>
   <property name="menuRole">
    <enum>QAction::TextHeuristicRole</enum>
   </property>
  </action>
  <action name="action_save_file_2">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-save.png</normaloff>:/FPGresources/icons48/icons8-save.png
    </iconset>
   </property>
   <property name="text">
    <string>Сохранить</string>
   </property>
   <property name="toolTip">
    <string>Сохранить</string>
   </property>
  </action>
  <action name="action_about_program">
   <property name="icon">
    <iconset resource="resources.qrc">
        <normaloff>:/FPGresources/icons48/icons8-info.png</normaloff>:/FPGresources/icons48/icons8-info.png
    </iconset>
   </property>
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MText_list_Widget</class>
   <extends>QListWidget</extends>
   <header>MWidgets.MList_Widgets.MText_list_Widget</header>
  </customwidget>
  <customwidget>
   <class>MColor_list_Widget</class>
   <extends>QListWidget</extends>
   <header>MWidgets.MList_Widgets.MColor_list_Widget</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
